File: testers/combined/java/generated-and-checked-in/avro_events/combined/avro_events/consumer/MoneyHandler.java
package combined.avro_events.consumer;

import com.example.events.common.Money;
import combined.avro_events.header.StandardHeaders;
import java.lang.Void;
import java.util.concurrent.CompletableFuture;

/** Handler interface for money topic events */
public interface MoneyHandler {
  /** Handle a message from the topic */
  CompletableFuture<Void> handle(
    String key,
    Money value,
    StandardHeaders headers
  );
}

File: testers/combined/kotlin/generated-and-checked-in/avro_events/combined/avro_events/consumer/MoneyHandler.kt
package combined.avro_events.consumer

import com.example.events.common.Money
import combined.avro_events.header.StandardHeaders
import io.smallrye.mutiny.Uni

/** Handler interface for money topic events */
interface MoneyHandler {
  /** Handle a message from the topic */
  abstract fun handle(
    key: kotlin.String,
    value: Money,
    headers: StandardHeaders
  ): Uni<Unit>
}